### ISSUE TITLE
Code of segmentation on MSCOCO

### ISSUE THREAD

**Issue by zhaozhengChen** [jbeomlee93/RIB#2]
Hi authors,
[deeplab-pytorch](https://github.com/kazuto1011/deeplab-pytorch) only support VOC and COCO-stuff.
Could you please release your code for segmentation on MSCOCO2014?
Thanks!

**Comment by PengtaoJiang** [comment]
@jbeomlee93, I have the same request. Could you provide the deeplab-pytorch code for MS COCO? I want to know the hyper-parameters in your experiments.

**Comment by jbeomlee93** [comment]
Dear @zhaozhengChen and @PengtaoJiang,

Sorry for the late reply.

We used the below configs for training a segmentation network.

    EXP:
        ID: coco
        OUTPUT_DIR: data_coco_RIB
    
    DATASET:
        NAME: coco
        ROOT: ./data/datasets/coco_2014
        LABELS: ./data_coco/datasets/coco/labels.txt
        N_CLASSES: 81
        IGNORE_LABEL: 255
        SCALES: [0.5, 0.75, 1.0, 1.25, 1.5, 1.75, 2.0]
        SPLIT:
            TRAIN: train_2014
            VAL: val_2014
            TEST: test
    
    DATALOADER:
        NUM_WORKERS: 8
    
    IMAGE:
        MEAN:
            R: 122.675
            G: 116.669
            B: 104.008
        SIZE:
            BASE: # None
            TRAIN: 481
            TEST: 513
    
    MODEL:
        NAME: DeepLabV2_ResNet101_MSC
        N_BLOCKS: [3, 4, 23, 3]
        ATROUS_RATES: [6, 12, 18, 24]
        INIT_MODEL: data/models/imagenet/deeplabv1_resnet101-imagenet.pth
    
    SOLVER:
        BATCH_SIZE:
            TRAIN: 10
            TEST: 1
        ITER_MAX: 100000
        ITER_SIZE: 1
        ITER_SAVE: 2500
        ITER_TB: 20
        LR_DECAY: 10
        LR: 2.5e-4
        MOMENTUM: 0.9
        OPTIMIZER: sgd
        POLY_POWER: 0.9
        WEIGHT_DECAY: 5.0e-4
        AVERAGE_LOSS: 20
        FREEZE_BN: True
        BALANCE_LOSS: True

    CRF:
        ITER_MAX: 10
        POS_W: 3
        POS_XY_STD: 1
        BI_W: 4
        BI_XY_STD: 67
        BI_RGB_STD: 3

We also used balanced cross-entropy loss. Please refer to [this](https://github.com/jbeomlee93/AdvCAM/issues/9#issuecomment-885420708).

Thanks!

**Comment by Italy2006** [comment]
> Dear @zhaozhengChen and @PengtaoJiang,
> 
> Sorry for the late reply.
> 
> We used the below configs for training a segmentation network.
> 
> ```
> EXP:
>     ID: coco
>     OUTPUT_DIR: data_coco_RIB
> 
> DATASET:
>     NAME: coco
>     ROOT: ./data/datasets/coco_2014
>     LABELS: ./data_coco/datasets/coco/labels.txt
>     N_CLASSES: 81
>     IGNORE_LABEL: 255
>     SCALES: [0.5, 0.75, 1.0, 1.25, 1.5, 1.75, 2.0]
>     SPLIT:
>         TRAIN: train_2014
>         VAL: val_2014
>         TEST: test
> 
> DATALOADER:
>     NUM_WORKERS: 8
> 
> IMAGE:
>     MEAN:
>         R: 122.675
>         G: 116.669
>         B: 104.008
>     SIZE:
>         BASE: # None
>         TRAIN: 481
>         TEST: 513
> 
> MODEL:
>     NAME: DeepLabV2_ResNet101_MSC
>     N_BLOCKS: [3, 4, 23, 3]
>     ATROUS_RATES: [6, 12, 18, 24]
>     INIT_MODEL: data/models/imagenet/deeplabv1_resnet101-imagenet.pth
> 
> SOLVER:
>     BATCH_SIZE:
>         TRAIN: 10
>         TEST: 1
>     ITER_MAX: 100000
>     ITER_SIZE: 1
>     ITER_SAVE: 2500
>     ITER_TB: 20
>     LR_DECAY: 10
>     LR: 2.5e-4
>     MOMENTUM: 0.9
>     OPTIMIZER: sgd
>     POLY_POWER: 0.9
>     WEIGHT_DECAY: 5.0e-4
>     AVERAGE_LOSS: 20
>     FREEZE_BN: True
>     BALANCE_LOSS: True
> 
> CRF:
>     ITER_MAX: 10
>     POS_W: 3
>     POS_XY_STD: 1
>     BI_W: 4
>     BI_XY_STD: 67
>     BI_RGB_STD: 3
> ```
> 
> We also used balanced cross-entropy loss. Please refer to [this](https://github.com/jbeomlee93/AdvCAM/issues/9#issuecomment-885420708).
> 
> Thanks!

Thanks for your work, but i want to know how many GPU you use for coco training? And how to use the freeze bn in the network?

**Comment by libingruirrr** [comment]
您好，我想问下，您的预训练权重deeplabv1_resnet101-imagenet.pth有下载链接吗
